stan/math/rev/mat/fun/adj_jac_apply.hpp

```diff
@@ -54,9 +261,88 @@ struct adj_jac_vari : public vari {
     }
   }
 
+  /*
+   * accumulate_adjoints recurses through the adjoints returned from the
+   * user-supplied adjoint_jacobian_multiply and accumulates the appropriate
+   * vari adjoints pointed to by values of x_vis_.
+   *
+   * The adjoints of the nth argument are accumulated into the adjoints of the
+   * varis pointed to in the range x_vis_ + offsets_[n] to x_vis_ + offsets[n] +
+   * size of nth argument.
+   *
+   * If is_var_[n] is false, then the nth argument is ignored, whatever its
+   * value is.
+   *
+   * Each of the arguments can be an Eigen::Matrix<double, R, C>, a
+   * std::vector<double>, a std::vector<int>, a double, or an int.
+   *
+   * As an example, take the call:
+   *    accumulate_adjoints(a1, a2, a3, a4, a5)
+   *
+   * If a1 is an Eigen::Matrix<double, 5, 5>, a2 is a double, a3 is a double,
+   * a4 is a length 5 std::vector of doubles, a5 is a length 3 std::vector of
+   * ints and is_var_ is set to {true, false, true, true, false}, the 25 values
+   * in a1 are accumulated in the adjs of the varis pointed to by the values of
+   * x_vis_ to x_vis_ + 24, the value of a2 is ignored, and the adjoint of the
+   * vari pointed to by x_vis_[25] is incremented by a3, the 5 values in a4 are
+   * accumulated in the adjs of the varis pointed to by the values of x_vis_ +
+   * 26 to x_vis_ + 30, and the values in a5 are ignored.
+   *
+   * @param y_adj_jac next set of adjoints to be accumulated
```

Comment:
Each of these functions should get doc including tparam.  

I can see why you want to skip it, as they're all just part of the same definition, but not from the C++ perspective where each one is completely independent as they're just overloads.